File: Python/Queries/MetaQuery.py
import re

# ============================================================
# get_query:
# params:
# ============================================================
def get_query(filename: str) -> str:

    try:
        
        with open(filename, 'r', encoding='utf-8') as file:
            
            return file.read()
            
    except FileNotFoundError:
        
        return f"Error:'{filename}' does not exist in directory."
# ============================================================

# ============================================================
# get_TableName:
# params:
# ============================================================
def get_TableName(query: str) -> str:

    m = match.group(1) if (match := re.search(r"FROM\s+(\w+)", query)) else None

File: Julia/Queries/MetaQuery.jl
module MetaQuery

    export get_query, table_name


    """
        get_query(filename::AbstractString) -> String

    Returns the SQL query from an SQL file as a String:
    """
    function get_query(filename::AbstractString)::String
        
        if !isfile(filename)
            throw(ArgumentError("file '$filename' not found"))
        end

        query = open(filename) do file
            read(file, String)
        end

        return query
    
    end


    """
        table_name(query::String) -> String

    Returns the table name from an SQL query:
    """
    function table_name(query::String)::String

        m = match(r"FROM\s+'(\w+)'", query).captures[1]
    
        if m === nothing
            throw(ArgumentError("""Unable to locate the 
                                   table name in your SQL query 
                                """
                  )
            )
        end
    
        return m

    end
    #****************************************************************

end
